src/main/kotlin/com/openlattice/shuttle/MissionControl.kt

```diff
@@ -350,11 +350,23 @@ class MissionControl(
                             mutableSetOf(),
                             organizationId
                     )
-                    val entitySetId = entitySetsApi
-                            .createEntitySets(setOf(entitySet))
-                            .getValue(entityDefinition.entitySetName)
-                    check(entitySetId == entitySet.id) { "Submitted entity set id does not match return." }
-                    entitySets[entityDefinition.entitySetName] = entitySet
+                    entitySets[entityDefinition.entitySetName] = try {
+                        entitySetsApi
+                                .createEntitySets(setOf(entitySet))
+                                .getValue(entityDefinition.entitySetName)
+                        entitySetsApi
+                                .getEntitySetsByName(setOf(entityDefinition.entitySetName))
+                                .getValue(entityDefinition.entitySetName)
+                    } catch (ex: RhizomeRetrofitCallException) {
+                        if (ex.body.contains("Type ${entityDefinition.entitySetName} already exists.")) {
```

Comment:
I'm confused as to how this is ever happening b/c line 341: `if (!entitySets.containsKey(entityDefinition.entitySetName)) {` ?

Reply:
Oh, I guess if other integrations create during this one or something

Reply:
Yeah, if you are running integrations in parallel locally which admittedly rare. I probably wouldn't have even done this PR to handle the small corner case of parallel local integrations, but I was a troubleshooting a separate bug (which turned out to be a permissions issue from user synchronization changes) that caused entity set creation to succeed but then integrations would fail with 401.

No objections to just closing out this PR without merging.